orttraining/orttraining/models/runner/training_runner.cc

```diff
@@ -388,14 +397,92 @@ Status TrainingRunner::TrainingLoop(IDataLoader& training_data_loader, IDataLoad
     epoch++;
   }
 
+  const size_t number_of_batches = step_ - step_start;
+  const size_t weight_update_steps = weight_update_step_count_ - weight_update_step_count_start;
+  const double avg_time_per_batch = total_time / (step_ - step_start) * 1000;
+  const double throughput = params_.batch_size * (step_ - step_start) / total_time;
+
+  if (params_.perf_output_dir.empty()) {
+    printf("No perf output directory specified, skipping save of trained perf metrics.\n");
+  } else {
+    ORT_RETURN_IF_ERROR(Env::Default().CreateFolder(params_.perf_output_dir));
+    // saving json file
+    ORT_RETURN_IF_ERROR(SavePerfMetrics(number_of_batches, gradient_accumulation_step_count, weight_update_steps, 
+                                        total_time, avg_time_per_batch, throughput));
+  }
+
   std::cout << "Round: " << round_ << "\n"
             << "Batch size: " << params_.batch_size << "\n"
-            << "Number of Batches: " << (step_ - step_start) << "\n"
+            << "Number of Batches: " << number_of_batches << "\n"
             << "Gradient Accumulation Steps: " << gradient_accumulation_step_count << "\n"
-            << "Weight Update Steps: " << (weight_update_step_count_ - weight_update_step_count_start) << "\n"
+            << "Weight Update Steps: " << weight_update_steps << "\n"
             << "Total Running Time: " << total_time << " Seconds \n"
-            << "Average Running Time Per Batch: " << total_time / (step_ - step_start) * 1000 << " ms\n"
-            << "Throughput: " << params_.batch_size * (step_ - step_start) / total_time << " Examples / Second\n";
+            << "Average Running Time Per Batch: " << avg_time_per_batch << " ms\n"
+            << "Throughput: " << throughput << " Examples / Second\n";
+
+  return Status::OK();
+}
+
+Status TrainingRunner::SavePerfMetrics(const size_t number_of_batches, const size_t gradient_accumulation_steps,
+                                       const size_t weight_update_steps, const double total_time,
+                                       const double avg_time_per_batch, const double throughput) {
+  // popualte metrics for reporting
+  json j;
+  j["Model"] = params_.model_type;  
+  auto it = params_.perf_properties.find("sequence");
+  std::string seq_len;
```

Comment:
Agreed with @edgchen1 comments - can we add these model-specific parameters to `params_` that the caller provides?  E.g. in `bert/main.cc` we can provide a default json object or `std::map<string, string>` or a callback to add extra properties.